pkg/internal/testing/process/arguments.go

```diff
@@ -3,11 +3,13 @@ package process
 import (
 	"bytes"
 	"html/template"
+	"sort"
+	"strings"
 )
 
 // RenderTemplates returns an []string to render the templates
 //
-// Deprecated: this should be removed when we remove APIServer.Args.
+// Deprecated: will be removed in favor of Arguments
```

Comment:
Ah!  It's a bit less straightforward (wasn't actually certain how commonly people used the templates beyond the default args -- haven't seen a lot of it), and maybe I could make it more straightforward, but it was intended that you could do something like:

```go
// in your code
type CertDirArg struct {
  server *APIServer
}
func (a CertDirArg) Append(...string) Arg { return a } // ignore appends

func (a CertDirArg) Get(defaults []string) []string { 
  return []string{filepath.Join(a.server.CertDir, "blah")}
}

// then later

server.Configure().SetRaw("--proxy-client-key-file", CertDirArg{server})
```

Then, you can do ``

On option to make this easier would be to add a helper to CR:

```go
// in CR:
type FuncArg func(defaults []string) []string
func (a FuncArg) Append(...string) Arg { return a } // ignore appends
func (a FuncArg) Get(defaults []string) []string {
  return a(defaults)
}

// in your code

server.Configure().SetRaw("--proxy-client-key-file", envtest.FuncArg(func([]string) {
  return []string{filepath.Join(server.CertDir, "blah")} // theoretically works on windows, too!
})
```

Finally, if you're just after the cert dir, and are using the pre-existing client CA certs as a nice generated set of certs, you could wrap the auth to have your own certs instead:

```go
type WithProxyCA struct {
  // delegate most stuff to the other authn
  envtest.Authn

  CA envtest.TinyCA
  CertDir string
}

func (a *WithProxyCA) Configure(workDir string, args, *process.Arguments) error {
  a.CertDir = workDir
  args.Set("--proxy-client-cert-file", filepath.Join(a.CertDir, "proxy-client.crt"))
  // etc
  return nil
}
func (a *WithProxyCA) Start() error {
  // write stuff out to disk
  a.CA.DoTheStuffYouWant()
  
  // delegated to the other authn stuff
  return a.Other.Start()
}
```

Reply:
Do any of these seem reasonable?

Reply:
Yes, good point. `SetRaw` would still support such use cases.

You're right, that's probably not straightforward / obvious how to do it (at least I didn't recognize it directly). I like the `FuncArg` idea. I think, if you add this (with a nice example like the above), it would already make things a bit more obvious. 👍 

> Finally, if you're just after the cert dir, and are using the pre-existing client CA certs as a nice generated set of certs, you could wrap the auth to have your own certs instead:

Yeah, that's actually what we're doing there. However, my comment was indeed rather about the general case and not tied to the special one :)